s3_upload/utils/upload.py

```diff
@@ -227,34 +280,54 @@ def multi_core_upload(
     -------
     dict
         mapping of local file to ETag ID of uploaded file
+    list
+        list of any files that failed to upload
     """
     log.info(
-        f"Beginning upload {len(files)} files with {cores} cores to"
-        f"{bucket}:{remote_path}"
+        "Beginning uploading %s files with %s cores to%s:%s",
+        len(files),
+        cores,
+        bucket,
+        remote_path,
     )
 
-    uploaded_files = {}
-
-    with ProcessPoolExecutor(max_workers=cores) as exe:
-        concurrent_jobs = {
-            exe.submit(
-                multi_thread_upload,
-                files=sub_files,
-                bucket=bucket,
-                remote_path=remote_path,
-                threads=threads,
-                parent_path=parent_path,
-            ): sub_files
-            for sub_files in files
-        }
+    all_uploaded_files = {}
+    all_failed_upload = []
+
+    with ProcessPoolExecutor(max_workers=cores) as executor:
+        concurrent_jobs = _submit_to_pool(
+            pool=executor,
+            func=multi_thread_upload,
+            item_input="files",
+            items=files,
+            bucket=bucket,
+            remote_path=remote_path,
+            parent_path=parent_path,
+            threads=threads,
+        )
 
         for future in as_completed(concurrent_jobs):
             # access returned output as each is returned in any order
             try:
-                uploaded_files = {**uploaded_files, **future.result()}
+                uploaded_files, failed_upload = future.result()
+
+                all_uploaded_files = {**all_uploaded_files, **uploaded_files}
```

Comment:
Could you please explain this line? I see it in other places in the script and I thought I knew what it did but I think I'm confusing myself